src/lib.rs

```diff
@@ -1385,6 +1391,121 @@ where
     Some(Ratio::new(n1, d1))
 }
 
+#[cfg(all(feature = "bigint", feature = "std"))]
+impl<T: Clone + Integer + Signed + ToPrimitive + ToBigInt> ToPrimitive for Ratio<T> {
+    fn to_i64(&self) -> Option<i64> {
+        (self.numer.clone() / self.denom.clone()).to_i64()
```

Comment:
These integer conversions can all be `self.to_integer().to_X()`, though we might want to also improve that to not go through `trunc()`, to avoid creating the 1 denominator at all. i.e. `trunc()` should call `to_integer()` instead of the other way around.